packages/libsql-client/src/http.ts

```diff
@@ -99,33 +102,35 @@ export class HttpClient implements Client {
     }
 
     async execute(stmt: InStatement): Promise<ResultSet> {
-        try {
-            const isSchemaDatabasePromise = this.getIsSchemaDatabase();
-            const hranaStmt = stmtToHrana(stmt);
-
-            // Pipeline all operations, so `hrana.HttpClient` can open the stream, execute the statement and
-            // close the stream in a single HTTP request.
-            let rowsPromise: Promise<hrana.RowsResult>;
-            const stream = this.#client.openStream();
+        return limit(async () => {
```

Comment:
This should probably be added to `batch`, `transaction` and `executeMultiple` as well.
We should do the same in the web socket client.

Reply:
Added! The only issue is that we lost type safety while adding all these calls to a single queue:

```
this.#limit = promiseLimit<any>(concurrency);
```

This means that if the return type doesn't match, we won't get a TypeScript error, although the code still works.

An alternative would be to keep 4 queues so each function has its own queue (`execute`, `batch`, `transaction`, and `executeMultiple`), but for now we are keeping one.

I filled an issue in the [promise-limit](https://github.com/featurist/promise-limit/issues/23) repo to see if we can get support for types on a per-call basis.